gauge-browser-agent-otel/tests/step_implementation.js

```diff
@@ -15,64 +15,68 @@ const {
 } = require("taiko");
 const assert = require("assert");
 const headless = process.env.headless_chrome.toLowerCase() === "true";
-const config = require("../config");
-const jwtDecode = require("jwt-decode");
-const axios = require("axios");
+const SLService = require("../services/sealightsService");
 
-const decoded = jwtDecode(config.apiToken); // Agent Token
-const baseUrl = decoded["x-sl-server"]; // Base url of the backend
+let testSession;
+let testStartTime;
 
-const createTestSession = () => {
-  return axios.post(
-    baseUrl.replace("/api", "/sl-api/v1/test-sessions"), // Public API to start a test session
-    {
-      testStage: "Gauge Tests",
-      bsid: "51721270-ead5-498b-b22f-c3f9861ea44e", // Better set from environment variable or get from window.$Sealights
-    },
-    {
-      headers: {
-        Authorization: `Bearer ${config.apiToken}`,
-      },
-    }
-  );
-};
-
-// Works the same with any test framework, before test starts emit event to set baggage
-beforeScenario(async (scenario) => {
+beforeSuite(async () => {
   // Start a test session
-  const { testSessionId } = (await createTestSession()).data.data;
+  const { testSessionId } = (await SLService.createTestSession()).data.data;
+  testSession = testSessionId;
+
+  await openBrowser({
+    headless,
+    ignoreCertificateErrors: true,
+    args: ["--disable-web-security"], // CORS
+  });
+  await goto("http://localhost:3333");
+});
 
+beforeScenario(async (scenario) => {
+  // Set the correct baggage before a scenario runs with testName and the current testSessionId
   await evaluate(
     "",
     async (element, args) => {
       const testName = args.scenario.currentScenario.name;
       const customEvent = new CustomEvent("set:baggage", {
         detail: {
           "x-sl-test-name": testName,
-          "x-sl-test-session-id": args.testSessionId,
+          "x-sl-test-session-id": args.testSession,
         },
       });
       window.dispatchEvent(customEvent);
     },
-    { args: { scenario, testSessionId } }
+    { args: { scenario, testSession } }
   );
+  testStartTime = Date.now();
 });
 
-beforeSuite(async () => {
-  await openBrowser({
-    headless,
-    ignoreCertificateErrors: true,
-    args: ["--disable-web-security"], // CORS
+afterScenario(async (scenario) => {
+  // Unset baggage after scenario
+  await evaluate("", async () => {
+    const customEvent = new CustomEvent("delete:baggage");
+    window.dispatchEvent(customEvent);
   });
-  await goto("http://localhost:3333");
+  // Send test event to Sealights
+  const data = await SLService.sendTestEvent(
+    testSession,
+    scenario.currentScenario.name,
+    testStartTime,
+    Date.now(),
+    scenario.isFailed ? "failed" : "passed" // could also be "skipped" value, implement logic as needed
```

Comment:
Can we please add the skipped as well?

Reply:
I tried to but looks like there isn't a way to skip tests in `Gauge`, at least not in their docs... so I remove this comment as it is misleading. The only way to skip a test is if some error was thrown (like step wasn't implemented) but in that case it won't even execute any of the hooks.

Still not implemented as per this: https://github.com/getgauge/gauge/issues/1397
There is no way currently to skip execution at run time.

Reply:
This is out of scope for this task and was not in the DoD.